CondTools/SiStrip/plugins/SiStripCondVisualizer.cc

```diff
@@ -0,0 +1,532 @@
+#define LOGINFO(x) edm::LogInfo(x)
+#define LOGVERBATIM(x) edm::LogVerbatim(x)
+#define LOGDEBUG(x) LogDebug(x)
+
+// ROOT includes
+#include "TNamed.h"
+#include "TObjString.h"
+#include "TText.h"
+#include "TTree.h"
+
+//
+// Auxiliary enum to enumerate the conditions types
+//
+namespace SiStripCondTypes {
+  enum condformat { Noise = 0, Pedestals = 1, G1Gain = 2, G2Gain = 3, Quality = 4, EndOfTypes = 99 };
+  static const std::array<std::string, 5> titles = {{"noise", "pedestals", "G1 gain", "G2 gain", "Quality"}};
+  static const std::array<std::string, 5> units = {{"[ADC counts]", "[ADC counts]", "", "", ""}};
+
+  // some magic from https://stackoverflow.com/questions/47801709/best-way-to-set-a-bitset-with-boolean-values
+  template <typename... Args>
+  std::bitset<sizeof...(Args)> makeBitSet(Args... as) {
+    using unused = bool[];
+    std::bitset<sizeof...(Args)> ret;
+    std::size_t ui{ret.size()};
+    (void)unused{true, (ret.set(--ui, as), true)...};
+    return ret;
+  }
+
+}  // namespace SiStripCondTypes
+
+//
+// class declaration
+//
+
+using HistoMap = std::map<uint32_t, TH1F*>;
+
+class SiStripCondVisualizer : public edm::one::EDAnalyzer<edm::one::SharedResources, edm::one::WatchRuns> {
+public:
+  explicit SiStripCondVisualizer(const edm::ParameterSet&);
+  ~SiStripCondVisualizer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  void endRun(edm::Run const&, edm::EventSetup const&) override{};
+  bool isDetIdSelected(const uint32_t detid);
+  HistoMap bookModuleHistograms(const TrackerTopology* tTopo, const SiStripCondTypes::condformat& type);
+  std::tuple<std::string, int, int, int> setTopoInfo(uint32_t detId, const TrackerTopology* tTopo);
+  std::string module_location_type(const unsigned int& mod);
+  void fillTheQualityMap(const SiStripQuality* obj, HistoMap& theMap);
+  template <class Payload>
+  void fillTheHistoMap(const Payload* obj, HistoMap& theMap);
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  // ----------member data ---------------------------
+  edm::ESGetToken<SiStripNoises, SiStripNoisesRcd> noiseToken_;
+  edm::ESGetToken<SiStripPedestals, SiStripPedestalsRcd> pedToken_;
+  edm::ESGetToken<SiStripApvGain, SiStripApvGainRcd> g1Token_;
+  edm::ESGetToken<SiStripApvGain, SiStripApvGain2Rcd> g2Token_;
+  edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualToken_;
+  const edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> topoToken_;
+
+  const bool doNoise_;
+  const bool doPeds_;
+  const bool doG1_;
+  const bool doG2_;
+  const bool doBadComps_;
+
+  std::vector<DetIdSelector> detidsels_;
+  edm::Service<TFileService> fs;
+  SiStripDetInfo detInfo_;
+  std::map<std::string, TFileDirectory> outputFolders;
+  HistoMap NoiseMap_, PedMap_, G1Map_, G2Map_, QualMap_;
+  std::bitset<5> plottedConditions;
```

Comment:
should these be unified to all have the trailing underscore?

Reply:
not really relevant. Isn't it?